src/Behat/Testwork/ServiceContainer/ExtensionManager.php

```diff
@@ -228,4 +234,29 @@ private function validateExtensionInstance($extension, $locator)
             ), $locator);
         }
     }
+
+    private function discoverExtensionWithPuli($locator)
```

Comment:
I'd rather this method and all the other mentions of Puli to be hidden behind an abstraction.